src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  function add(n) {
+    calculator.result += n;
+  }
+
+  function subtract(n) {
+    calculator.result -= n;
+  }
+
+  function multiply(n) {
+    calculator.result *= n;
+  }
+
+  function divide(n) {
+    calculator.result /= n;
+  }
+
+  function reset() {
+    calculator.result = 0;
+
+    return this;
+  }
+
+  function operate(callback, n) {
+    if (typeof callback === 'function') {
+      callback(n);
+    }
+
+    return this;
```

Comment:
[CODE KNOWLEDGE] For the `operate` method, the method shorthand syntax should be used to keep the object definition concise and consistent.

BAD EXAMPLE:
 operate: function(callback, n) {
   if (typeof callback === 'function') {
     callback(n);
   }
   return this;
 },

GOOD EXAMPLE:
 operate(callback, n) {
   if (typeof callback === 'function') {
     callback(n);
   }
   return this;
 },